type-c-service/src/service/ucsi.rs

```diff
@@ -165,9 +235,46 @@ impl<'a> Service<'a> {
                     response.notify_opm = false;
                     response.cci = Cci::default();
                     response.data = Ok(None);
+                    self.set_cci_connector_change(state, &mut response.cci);
                     return response;
                 }
             }
         }
     }
+
+    pub(super) async fn process_ucsi_event(&self, port_id: GlobalPortId, port_event: PortStatusChanged) {
+        let state = &mut self.state.lock().await.ucsi;
+        let mut ucsi_event = ConnectorStatusChange::default();
+
+        ucsi_event.set_connect_change(port_event.plug_inserted_or_removed());
+        ucsi_event.set_power_direction_changed(port_event.power_swap_completed());
+        ucsi_event.set_pd_reset_complete(port_event.pd_hard_reset());
+
+        if port_event.data_swap_completed() || port_event.alt_mode_entered() {
+            ucsi_event.set_connector_partner_changed(true);
+        }
+
+        if port_event.new_power_contract_as_consumer() || port_event.new_power_contract_as_provider() {
+            ucsi_event.set_negotiated_power_level_change(true);
+            ucsi_event.set_power_op_mode_change(true);
+            ucsi_event.set_external_supply_change(true);
+            ucsi_event.set_power_direction_changed(true);
+            ucsi_event.set_battery_charging_status_change(true);
+        }
+
+        if !ucsi_event.filter_enabled(state.notifications_enabled).is_none() {
+            state.pending_ports.pend_port(port_id.0 as usize);
+            debug!(
+                "Port{}: Queuing UCSI connector change event: {:?}",
+                port_id.0, ucsi_event
+            );
+        }
+
+        if state.pending_ports_iter.is_none() && !state.pending_ports.is_none() {
+            // Start a new round robin pass and broadcast the initial event
+            // Subsequent events will be broadcast when ACK_CC_CI is processed
+            self.start_connector_changed_notify(state, state.notifications_enabled.connect_change())
+                .await;
```

Comment:
Why can't we publish all of the notifications in one pass? Managing a partial notification adds complexity and I don't see why we need it.